src/ext/java/org/opentripplanner/ext/fares/impl/DefaultFareService.java

```diff
@@ -350,34 +322,36 @@ private List<Leg> combineInterlinedLegs(List<Leg> fareLegs) {
   }
 
   private FareSearch performSearch(
+    Currency currency,
     FareType fareType,
-    List<Leg> rides,
+    List<Leg> legs,
     Collection<FareRuleSet> fareRules
   ) {
-    FareSearch r = new FareSearch(rides.size());
+    FareSearch r = new FareSearch(legs.size());
+    final FareAndId UNKNOWN = new FareAndId(
+      Money.ofFractionalAmount(currency, 999f),
+      new FeedScopedId("max", "max")
+    );
 
     // Dynamic algorithm to calculate fare cost.
     // This is a modified Floyd-Warshall algorithm, a key thing to remember is that
-    // rides are already edges, so when comparing "via" routes, i -> k is connected
+    // legs are already edges, so when comparing "via" routes, i -> k is connected
     // to k+1 -> j.
-    for (int i = 0; i < rides.size(); i++) {
+    for (int i = 0; i < legs.size(); i++) {
       // each diagonal
-      for (int j = 0; j < rides.size() - i; j++) {
-        FareAndId best = getBestFareAndId(fareType, rides.subList(j, j + i + 1), fareRules);
-        float cost = best.fare;
-        if (cost < 0) {
-          LOG.error("negative cost for a ride sequence");
-          cost = Float.POSITIVE_INFINITY;
-        }
-        if (cost < Float.POSITIVE_INFINITY) {
+      for (int j = 0; j < legs.size() - i; j++) {
+        FareAndId best = getBestFareAndId(fareType, legs.subList(j, j + i + 1), fareRules)
+          .orElse(UNKNOWN);
+        Money cost = best.fare();
+        if (!cost.equals(UNKNOWN.fare())) {
           r.endOfComponent[j] = j + i;
           r.next[j][j + i] = j + i;
         }
         r.resultTable[j][j + i] = cost;
```

Comment:
Unknown fare is assigned to the result table here, as if it was a real price.  populateFare method does not seem to track the special value in any way but puts it to created itineraryFares, I think.

Reply:
Do we ever want to have a case there we return a price of -1 cents? It seems that it's wrong and we should not return any price/component at all, isn't it?

Reply:
Yes, performSearch() is not very self-documenting code. I have never really tried to understand it.